internal/command/cmd.go

```diff
@@ -45,18 +46,26 @@ const (
 // Returns error if all retries fail, nil otherwise.
 func (c *Cmd) RetryCommand(numOfRetries int) error {
 	var err error
+
+	// Have a cmd that is safe for printing.
+	printSafeCmd := utils.SanitiseKubeconfig(c.Command)
```

Comment:
I am actually not sure about this. The command package is used for any command, not just kubectl. Therefore, I do not see a reason to have this function call here. 

What I would do instead, is maybe something like 
```go
func (c *Cmd) RetryCommand(numOfRetries int) error {
    printSafeCmd := c.sanitiseCommand()
    //.....
}

// sanitiseCommand returns c.Command with all secrets removed by ***
func (c *Command)sanitiseCommand() string {
    var s string
    //in case c.Command is kubectl, remove any possible kubeconfig.
    s = utils.SanitiseKubeconfig(c.Command)
    //in case c.Command is XXX, remove YYY
    s = utils.SanitiseXXX(s)
    //.....
    return s
}
```

Reply:
got it, I see what you're saying.
I'll rework it, then.
to check for the actual cmd, we can just query arg[0] of `c.Args` and switch on it.

Reply:
>to check for the actual cmd, we can just query arg[0] of c.Args and switch on it.
yes, for the actual command, we will call `c.Command`, while sanitised will be returned from the function.. we can even add it to the `Command` struct, but I will leave this up to you.